Sources/SwiftDocC/Model/Identifier.swift

```diff
@@ -66,6 +67,78 @@ public enum TopicReferenceResolutionResult: Hashable, CustomStringConvertible {
     }
 }
 
+/// The error causing the failure in the resolution of a ``TopicReference``.
+public struct TopicReferenceResolutionError: Error {
+    // we store the base as an `Error` so that
+    // we can potentailly pass through a `DescribedError`
+    // conformance in the future
+    private let base: Error
```

Comment:
If the message and note was stored directly instead of storing the base error then this struct can conform to `DescribedError`. It also makes it easier to conform to `Hashable` (although I think `Solution` and `Replacement` would need to explicitly declare conformance as well).

The functionality of accepting any error as input can still be achieved with an initializer that takes an error argument and reads the `localizedDescription` and `recoverySuggestion`.

Reply:
I added all four properties of the `DescribedError` protocol as stored constants to the error so I don't loose information when passing though an error. `Hashable` conformance is better now 👍